velox/functions/remote/client/CMakeLists.txt

```diff
@@ -16,11 +16,23 @@ velox_add_library(velox_functions_remote_thrift_client ThriftClient.cpp)
 velox_link_libraries(velox_functions_remote_thrift_client
                      PUBLIC remote_function_thrift FBThrift::thriftcpp2)
 
+set(curl_SOURCE BUNDLED)
+velox_resolve_dependency(curl)
```

Comment:
This doesn't actually build curl. We only use it to force cpr to use the version we want. So if you require curl I would say use `set_source` to allow system curl to be used (this might require changes to cpr.cmake but iirc cpr can work with system curl as well). 

Also move this into the root cml within an if().